### ISSUE TITLE
PolyPen: Unhandled Exception Caught!

### ISSUE THREAD

**Issue by naptownhero** [CGCookie/retopoflow#585]
I had multiple islands and switched from contour to pen. Used the pen and crash.

-------------------------------------

```
Environment:

- RetopoFlow: 2.0.0
- Blender: 2.79.0 master 2018-03-22
- Platform: Windows, 10, 10.0.17134, AMD64, Intel64 Family 6 Model 94 Stepping 3, GenuineIntel
- GPU: Intel, Intel(R) HD Graphics 530, 4.5.0 - Build 24.20.100.6194, 4.50 - Build 24.20.100.6194
- Timestamp: 2018-08-04 21:30:36.260767


Error Hash: 9a42177a85e4c84a725eab0bc42f53aa


Trace:

- EXCEPTION (<class 'ValueError'>): edges.new(): this edge exists
- .../rfmode\rfmesh.py
- 000 1331:new_edge() bme = self.bme.edges.new(verts)
- .../rfmode\rfcontext_target.py
- 001 0396:new_edge() return self.rftarget.new_edge(verts)
- .../rfmode\rftool_polypen.py
- 002 0282:modal_insert() bme = self.rfcontext.new_edge((bmv0, bmv1))
- .../rfmode\rftool.py
- 003 0076:wrapper() ret = fn(*args, **kwargs)
- 004 0110:modal() nmode = self.FSM[self.mode]()
- 005 0114:modal() raise e     # passing on the exception to RFContext
- .../rfmode\rfcontext.py
- 006 0556:modal_main() self.tool.modal()
- 007 0469:modal() nmode = self.FSM[self.mode]()
```

**Comment by vxlcoder** [comment]
Thanks for the report @naptownhero.
Could you provide a screenshot of what you did or steps to reproduce this?

**Comment by vxlcoder** [comment]
This is the same issue as #603, and it should now be fixed in the latest version of `release/2.0.1` branch. 
 If you know how, feel free to test that version, otherwise it will be officially released soon.

I'm closing this issue, but reopen if you continue to have problems.

Thanks for reporting this issue!
